apps/explorer/src/app/components/txs/details/tx-batch.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { t } from '@vegaprotocol/react-helpers';
+import type {
+  BlockExplorerTransactionResult,
+  BatchMarketInstructions,
+} from '../../../routes/types/block-explorer-response';
+import type { TendermintBlocksResponse } from '../../../routes/blocks/tendermint-blocks-response';
+import { TxDetailsShared } from './shared/tx-details-shared';
+import { TableWithTbody, TableRow, TableCell } from '../../table';
+
+interface TxDetailsBatchProps {
+  txData: BlockExplorerTransactionResult | undefined;
+  pubKey: string | undefined;
+  blockData: TendermintBlocksResponse | undefined;
+}
+
+/**
+ * BatchMarketInstructions are sets of changes in three categories: new orders (submissions),
+ * order change s(amendments) and cancellations.
+ *
+ * Design considerations for batch:
+ * - So far it's very basic details about the size of the batch
+ */
+export const TxDetailsBatch = ({
+  txData,
+  pubKey,
+  blockData,
+}: TxDetailsBatchProps) => {
+  if (!txData) {
+    return <>{t('Awaiting Block Explorer transaction details')}</>;
+  }
+
+  const cmd = txData.command as BatchMarketInstructions;
```

Comment:
question (not sure): what's the type you get when you try to access the command prop on the txData? usually type casting like this with `as` indicates a potential leak

Reply:
dropping a comment below on the type interface, that seems to be the source i think

Reply:
Yeah the types in this PR are a little bit lame as they're manually written. I'm going to switch to types generated from the swagger file (https://raw.githubusercontent.com/vegaprotocol/documentation/main/specs/v0.62.1/blockexplorer.openapi.json) - I'll make a ticket for that ⌛